discover/discover.go

```diff
@@ -76,21 +76,28 @@ func NewPublicKeyRecord(key jwk.Key, issuer string) (*PublicKeyRecord, error) {
 
 func DefaultPubkeyFinder() *PublicKeyFinder {
 	return &PublicKeyFinder{
-		JwksFunc: GetJwksByIssuer,
+		JwksFunc:   GetJwksByIssuer,
+		HttpClient: nil,
 	}
 }
 
-type JwksFetchFunc func(ctx context.Context, issuer string) ([]byte, error)
+type JwksFetchFunc func(ctx context.Context, issuer string, httpClient *http.Client) ([]byte, error)
 
 type PublicKeyFinder struct {
-	JwksFunc JwksFetchFunc
+	JwksFunc   JwksFetchFunc
+	HttpClient *http.Client
```

Comment:
Do we need both of these?

If JwksFetchFunc takes httpClient as a parameter, then do we need HttpClient on the struct?
Alternatively if HttpClient is on the struct couldn't it just get HttpClient via a closure?

Reply:
Thanks! Fixed here: be253b5